app/assets/src/components/views/samples/SamplesView.jsx

```diff
@@ -154,6 +154,27 @@ class SamplesView extends React.Component {
     }
   }
 
+  // Queries the samples prop (an ObjectCollectionView) for the job
+  // status of each selected sample id. Returns a Map where the
+  // keys are sampleIds and values are the name and status of the sample.
+  getSelectedSampleStatuses = () => {
```

Comment:
Does this work if there are 1000 samples on the page, and `ObjectCollectionView` hasn't loaded all of them yet? In that case, I think everything that's not yet loaded will be marked as `loading`.

Unfortunately, there is a way to select unloaded samples, and this is with the `Select All` checkbox. Select All is an important thing we need to support for ease-of-use. If there isn't an easy way to get the sample statuses here, you might need to make an API call i nBulkDownloadModal after all.

Reply:
You're right, I had to rewrite things and add a route on the backend.